### PR TITLE
Add DeepLC recipe

### PR DESCRIPTION
**Pull request by RalfG**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

 * `@BiocondaBot please update` will cause the BiocondaBot to merge the master branch into a PR
 * `@BiocondaBot please add label` will add the `please review & merge` label.
 * `@BiocondaBot please fetch artifacts` will post links to packages and docker containers built by the CI system. You can use this to test packages locally before merging.

For members of the Bioconda project, the following command is also available:

 * `@BiocondaBot please merge` will cause packages/containers to be uploaded and a PR merged. Someone must approve a PR first! This has the benefit of not wasting CI build time required by manually merging PRs.

If you have questions, please post them in gitter or ping `@bioconda/core` in a comment (if you are not able to directly ping `@bioconda/core` then the bot will repost your comment and enable pinging).

### CODE REVIEW

**Comment by npavlovikj** [comment]
The error is:
```
pkg_resources.DistributionNotFound: The 'xgboost<2,>=0.90' distribution was not found and is required by deeplc
```

You should add `xgboost >=0.90,<2` in the `run` section.

**Comment by RalfG** [comment]
It's written as `xgboost >=0.90,<2` in the meta.yaml. I think the problem is with the fact that `xgboost` is only in conda-forge. In anaconda it's named `py-xgboost`, so I'll try that one. Altneratively, is there a way to tell conda to also check the conda-forge channel?

**Comment by RalfG** [comment]
This also didn't work... To summarize:

**With `xgboost` in meta.yaml host and run:**
Build and test environments setup by conda:
```
py-xgboost:	0.90-py37_4		conda-forge
xgboost:	0.90-py37he1b5a44_4	conda-forge
```

Error message after test:
```
pkg_resources.DistributionNotFound: The 'xgboost<2,>=0.90' distribution was not found and is required by deeplc
```

**With `py-xgboost` in meta.yaml host and run:**
Build and test environments setup by conda:
```
py-xgboost:	0.90-py37_4		conda-forge
```

Error message after test: 
```
pkg_resources.DistributionNotFound: The 'xgboost<2,>=0.90' distribution was not found and is required by deeplc
```

**setup.py looked like this in both cases:**
```
install_requires=[
    'setuptools>=42.0.1',
    'tensorflow>=1.14.0,<3',
    'xgboost>=0.90,<2',
    'scipy>=1.3.1,<2',
    'matplotlib>=3,<4',
    'numpy>=1.17,<2',
    'pandas>=0.25,<1',
],
```

It seems to me that conda is looking for packages in conda-forge and that it only finds `xgboost` in there, which makes sense. The strange part is that it also adds `py-xgboost: 0.90-py37_4 conda-forge` to the environment by itself, even if I only put `xgboost>=0.90,<2` in the requirements. 

Then again, in a local build and test it only works when using `py-xgboost`. When using `xgboost` I get `conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {"xgboost[version='>=0.90,<2']"}`.

**Comment by npavlovikj** [comment]
@RalfG , the channels bioconda recipes are looking for dependencies are `conda-forge`, `bioconda`, and `defaults`, https://bioconda.github.io/user/install.html?highlight=channel#set-up-channels.

**Comment by RalfG** [comment]
@npavlovikj Thanks! For now we can actually do without XGBoost, so I just removed the dependency. If we do need it again, I'll look further into the issue.

**Comment by npavlovikj** [comment]
@RalfG , sounds good! I was just reading the error messages, haven't had the time to play with the recipe more :)

**Comment by RalfG** [comment]
@npavlovikj Now there's another dependency issue:
```
pkg_resources.ContextualVersionConflict: (opt-einsum 0+untagged.49.gdbede45.dirty (/opt/conda/conda-bld/deeplc_1576512229777/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.7/site-packages), Requirement.parse('opt-einsum>=2.3.2'), {'tensorflow'})
```
This arises during testing (which is just calling the `--help`). In the test environment, the following package is installed: `opt_einsum:    3.1.0-py_0    conda-forge`.

This is a tensorflow dependency. Any idea how to resolve this issue?

**Comment by npavlovikj** [comment]
@RalfG , if you explicitly add `opt-einsum >=2.3.2` in the `run` section, the recipe should build fine.

However, I just noticed that this is a dev version. Bioconda's practice is to have recipes for stable releases. For dev releases, you can upload packages to your own/lab conda channel. Since you are one of the developers of the package, is it possible to have a stable release instead, e.g. 0.1.1?

File: recipes/deeplc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+test:
+  imports:
+    - deeplc
+  commands:
+    - deeplc --help
+
+about:
+  home: http://compomics.github.io/projects/DeepLC
+  license: Apache Software
```
Comment (npavlovikj) [inline recipes/deeplc/meta.yaml]: Please remove the word `Software`, and use `Apache-2.0` instead.

File: recipes/deeplc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  license_family: APACHE
+  summary: "DeepLC: Retention time prediction for (modified) peptides using Deep Learning."
```
[the remark this answers is not in the document]
Comment (RalfG) [reply recipes/deeplc/meta.yaml]: Sure! For now I added the LICENSE file to this repo. In the next version I'll add it to the PyPI tarball.

**Comment by RalfG** [comment]
@BiocondaBot please add label

**Comment by npavlovikj** [comment]
@BiocondaBot please fetch artifacts

**Comment by npavlovikj** [comment]
@BiocondaBot please merge

**Comment by bgruening** [comment]
@BiocondaBot please merge

**Pull request merged**